### ISSUE TITLE
Nima on native-image - second request blocked forever

### ISSUE THREAD

**Issue by lost22git** [helidon-io/helidon#5273]
|||
|---|---|
|OS|win11|
|JDK|GraalVM CE 23.0.0-dev (build 19+36-jvmci-23.0-b01)|
|Helidon|4.0.0-alpha2|

The second request will be blocking when using Helidon Nima native image.

![image](https://user-images.githubusercontent.com/65008815/198327614-b4d426b1-1c92-46d0-9732-941a05d907ed.png)

## Main
```java

public class Main {

  public static void main(String[] args) {

    // 启动完毕后退出 主要用来查看启动耗时
//    System.setProperty("exit.on.started", "!");

    // SE 下需要在运行入口处初始化 logging
    LogConfig.configureRuntime();

    configureJackson();

    var server = WebServer.builder()
      .socket("h1", (sb, rb) -> {
//        sb.port(8080); // overrides the config from app.properties
        var routing = HttpRouting.builder()
          .update(Main::configureRouting)
          .build();
        rb.addRouting(routing);
      })
      .start();
  }

  static ObjectMapper JSON;

  static void configureJackson() {
    JSON = ObjectMapperCustomFn.createDefault().apply(new ObjectMapper());
  }

  static void configureRouting(HttpRouting.Builder routing) {
    // noinspection unchecked
    routing.register("/info", InfoRoute::new);
  }

}
```
## Routing
```java
public class InfoRoute implements HttpService {

  @Override
  public void routing(HttpRules rules) {
    rules.get("", this::getInfo);
  }

  void getInfo(ServerRequest req,
               ServerResponse res) {
    var info = SystemInfo.create();
    res.headers().contentType(HttpMediaType.APPLICATION_JSON);
    res.send(info);
  }
}
```

## Jackson media support
```java
public class JacksonSupportProvider implements MediaSupportProvider {

  private static final System.Logger LOGGER = System.getLogger(JacksonSupportProvider.class.getName());

  <T> boolean checkSupported(GenericType<T> type,
                             Headers requestHeaders,
                             WritableHeaders<?> responseHeaders) {
    if (responseHeaders != null
        && responseHeaders.contentType().filter(v -> v.test(APPLICATION_JSON)).isPresent()) {
      return true;
    }
    return requestHeaders != null && requestHeaders.acceptedTypes().contains(APPLICATION_JSON);
  }

  @Override
  public <T> WriterResponse<T> writer(GenericType<T> type,
                                      WritableHeaders<?> requestHeaders) {

    return writer(type, requestHeaders, null);
  }

  @Override
  public <T> WriterResponse<T> writer(GenericType<T> type,
                                      Headers requestHeaders,
                                      WritableHeaders<?> responseHeaders) {
    LOGGER.log(TRACE, " writer for type=" + type.getTypeName());

    if (checkSupported(type, requestHeaders, responseHeaders)) {
      //noinspection unchecked
      return new WriterResponse<>(COMPATIBLE, () -> (EntityWriter<T>) JacksonEntityWriter.INSTANCE);
    }
    return WriterResponse.unsupported();
  }

  static class JacksonEntityWriter<T> implements EntityWriter<T> {

    private static final JacksonEntityWriter<?> INSTANCE = new JacksonEntityWriter<>();

    @Override
    public void write(GenericType<T> type,
                      T object,
                      OutputStream outputStream,
                      Headers requestHeaders,
                      WritableHeaders<?> responseHeaders) {
      try {
        Main.JSON.writeValue(outputStream, object);
      } catch (IOException e) {
        throw new RuntimeException(e);
      }
    }

    @Override
    public void write(GenericType<T> type,
                      T object,
                      OutputStream outputStream,
                      WritableHeaders<?> headers) {
      write(type, object, outputStream, headers, null);
    }
  }
}
```

**Comment by danielkec** [comment]
Cannot reproduce on Linux with Graal 22.3.0

```
kec@vulcan:~/idp/reproducers/nima/gh5273$ native-image --version
GraalVM 22.3.0 Java 19 CE (Java Version 19.0.1+10-jvmci-22.3-b08)
```

```
kec@vulcan:~/idp/reproducers/nima/gh5273$ ./target/gh5273 
2022.10.30 14:31:03 INFO io.helidon.logging.jul.JulProvider Thread[#1,main,5,main]: Logging at runtime configured using classpath: /logging.properties
2022.10.30 14:31:03 INFO io.helidon.nima.webserver.ServerListener VirtualThread[#51]/runnable@ForkJoinPool-6-worker-1: [0x7b83c1a1] http://127.0.0.1:8080 bound for socket 'h1'
2022.10.30 14:31:03 INFO io.helidon.nima.webserver.ServerListener VirtualThread[#51]/runnable@ForkJoinPool-6-worker-1: [0x7b83c1a1] direct writes
2022.10.30 14:31:03 INFO io.helidon.nima.webserver.LoomServer Thread[#1,main,5,main]: Helidon Níma 4.0.0-ALPHA2
2022.10.30 14:31:03 INFO io.helidon.nima.webserver.LoomServer Thread[#1,main,5,main]: Started all channels in 0 milliseconds. 47 milliseconds since JVM startup. Java 19.0.1+10-jvmci-22.3-b08
```

```
kec@vulcan:~$ curl http://127.0.0.1:8080/info
{"info":"This is info!"}kec@vulcan:~$ curl http://127.0.0.1:8080/info
{"info":"This is info!"}kec@vulcan:~$ curl http://127.0.0.1:8080/info
{"info":"This is info!"}kec@vulcan:~$ curl http://127.0.0.1:8080/info
{"info":"This is info!"}kec@vulcan:~$ curl http://127.0.0.1:8080/info
{"info":"This is info!"}kec@vulcan:~$ curl http://127.0.0.1:8080/info
{"info":"This is info!"}kec@vulcan:~$ curl http://127.0.0.1:8080/info
{"info":"This is info!"}kec@vulcan:~$ curl http://127.0.0.1:8080/info
{"info":"This is info!"}kec@vulcan:~$ curl http://127.0.0.1:8080/info
{"info":"This is info!"}kec@vulcan:~$ curl http://127.0.0.1:8080/info
{"info":"This is info!"}
```

Reproducer attempt: https://github.com/danielkec/reproducers/tree/master/nima/gh5273

**Comment by danielkec** [comment]
@lost22git 

- Can you share what `ObjectMapperCustomFn.createDefault().apply(new ObjectMapper())` does? 
- Do you use any special configuration?
- Can you check if our [reproducer](https://github.com/danielkec/reproducers/tree/master/nima/gh5273) differs from you project?

**Comment by lost22git** [comment]
@danielkec 
I attempted https://github.com/danielkec/reproducers/tree/master/nima/gh5273 and same problem occurred  
It is a problem of native image in windows ?🤔

![image](https://user-images.githubusercontent.com/65008815/198888253-f06d962e-5c7b-482f-9788-910e7f888150.png)

![image](https://user-images.githubusercontent.com/65008815/198888593-60b4f1f9-09b6-45ab-8e46-07759a633efc.png)

![image](https://user-images.githubusercontent.com/65008815/198888236-1e9e3096-4346-46a0-b855-7a379649bcf1.png)

**Comment by danielkec** [comment]
Huh after lot of fun with Windows I managed to build the native image. But I couldn't reproduce your problem.
![image](https://user-images.githubusercontent.com/1773630/199558030-8674c8e0-ab18-4167-b00b-1445d1c9365d.png)
![image](https://user-images.githubusercontent.com/1773630/199558173-5592172e-f672-43ad-82ce-eb6a62bf0eba.png)
It just works as expected

**Comment by danielkec** [comment]
Maybe different version of visual studio stuff, did your native image build thrown any warning? What was the c compiler version you had?
```
[INFO] --- helidon-maven-plugin:3.0.0:native-image (native-image) @ gh5273 ---
[INFO] Building native image :C:\Users\danie\IdeaProjects\reproducers\nima\gh5273\target\gh5273
[WARNING] Warning: Ignoring server-mode native-image argument --no-server.
[WARNING] WARNING: Using incubator modules: jdk.incubator.vector, jdk.incubator.concurrent
[INFO] ========================================================================================================================
[INFO] GraalVM Native Image: Generating 'gh5273' (executable)...
[INFO] ========================================================================================================================
[INFO] [1/7] Initializing...                                                                                    (6.6s @ 0.11GB)
[INFO]  Version info: 'GraalVM 22.3.0 Java 19 CE'
[INFO]  Java version info: '19.0.1+10-jvmci-22.3-b08'
[INFO]  C compiler: cl.exe (microsoft, x64, 19.29.30146)
[INFO]  Garbage collector: Serial GC
[INFO]  1 user-specific feature(s)
[INFO]  - io.helidon.integrations.graal.nativeimage.extension.HelidonReflectionFeature
[INFO] 2022.11.02 18:12:27 INFO io.helidon.logging.jul.JulProvider Thread[#1,main,5,main]: Logging at initialization configured using classpath: /logging.properties
[INFO] [2/7] Performing analysis...  [**********]                                                              (17.0s @ 0.79GB)
[INFO]    7,570 (87.74%) of  8,628 classes reachable
[INFO]    9,970 (57.97%) of 17,200 fields reachable
[INFO]   34,529 (55.02%) of 62,758 methods reachable
[INFO]      378 classes,   117 fields, and 1,598 methods registered for reflection
[INFO]       76 classes,    57 fields, and    68 methods registered for JNI access
[INFO]        5 native libraries: crypt32, ncrypt, psapi, version, winhttp
[INFO] [3/7] Building universe...                                                                               (2.9s @ 0.88GB)
[INFO] [4/7] Parsing methods...      [*]                                                                        (2.0s @ 0.81GB)
[INFO] [5/7] Inlining methods...     [***]                                                                      (1.1s @ 0.65GB)
[INFO] [6/7] Compiling methods...    [***]                                                                     (10.2s @ 0.54GB)
[INFO] [7/7] Creating image...                                                                                  (2.9s @ 0.60GB)
[INFO]   14.43MB (47.35%) for code area:    21,847 compilation units
[INFO]   15.74MB (51.67%) for image heap:  209,456 objects and 16 resources
[INFO]  303.44KB ( 0.97%) for other data
[INFO]   30.46MB in total
[INFO] ------------------------------------------------------------------------------------------------------------------------
[INFO] Top 10 packages in code area:                               Top 10 object types in image heap:
[INFO]    1.60MB sun.security.ssl                                     3.03MB byte[] for code metadata
[INFO]  928.34KB java.util                                            1.89MB java.lang.String
[INFO]  752.29KB java.lang.invoke                                     1.74MB java.lang.Class
[INFO]  720.47KB com.sun.crypto.provider                              1.74MB byte[] for general heap data
[INFO]  468.65KB sun.security.x509                                    1.39MB byte[] for java.lang.String
[INFO]  466.69KB io.helidon.config                                  650.55KB com.oracle.svm.core.hub.DynamicHubCompanion
[INFO]  416.15KB java.lang                                          365.75KB byte[] for reflection metadata
[INFO]  394.86KB java.io                                            363.05KB java.util.HashMap$Node
[INFO]  391.90KB java.util.concurrent                               351.30KB java.lang.String[]
[INFO]  275.72KB java.text                                          321.10KB java.lang.Object[]
[INFO]    7.96MB for 224 more packages                                3.50MB for 2075 more object types
[INFO] ------------------------------------------------------------------------------------------------------------------------
[INFO]                        12.5s (26.1% of total time) in 172 GCs | Peak RSS: 3.04GB | CPU load: 8.94
[INFO] ------------------------------------------------------------------------------------------------------------------------
[INFO] Produced artifacts:
[INFO]  C:\Users\danie\IdeaProjects\reproducers\nima\gh5273\target\gh5273.build_artifacts.txt (txt)
[INFO]  C:\Users\danie\IdeaProjects\reproducers\nima\gh5273\target\gh5273.exe (executable)
[INFO] ========================================================================================================================
[INFO] Finished generating 'gh5273' in 46.5s.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  51.323 s
[INFO] Finished at: 2022-11-02T18:13:07+01:00
[INFO] ------------------------------------------------------------------------
```

**Comment by lost22git** [comment]
### cl.exe version
```
Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30133 for x64
Copyright (C) Microsoft Corporation.  All rights reserved
```
### Build log
```
❯ mvn clean package -Pnative-image -e
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Detecting the operating system and CPU architecture
[INFO] ------------------------------------------------------------------------
[INFO] os.detected.name: windows
[INFO] os.detected.arch: x86_64
[INFO] os.detected.version: 10.0
[INFO] os.detected.version.major: 10
[INFO] os.detected.version.minor: 0
[INFO] os.detected.classifier: windows-x86_64
[INFO]
[INFO] ----------------< io.helidon.reproducers.gh5273:gh5273 >----------------
[INFO] Building gh5273 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ gh5273 ---
[INFO] Deleting E:\code\me\reproducers\nima\gh5273\target
[INFO]
[INFO] --- maven-resources-plugin:2.7:resources (default-resources) @ gh5273 ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ gh5273 ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 4 source files to E:\code\me\reproducers\nima\gh5273\target\classes
[INFO] /E:/code/me/reproducers/nima/gh5273/src/main/java/me/kec/se/quickstart/gh5273/JacksonSupportProvider.java: Some input files use unchecked or unsafe operations.
[INFO] /E:/code/me/reproducers/nima/gh5273/src/main/java/me/kec/se/quickstart/gh5273/JacksonSupportProvider.java: Recompile with -Xlint:unchecked for details.
[INFO]
[INFO] --- maven-resources-plugin:2.7:testResources (default-testResources) @ gh5273 ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory E:\code\me\reproducers\nima\gh5273\src\test\resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ gh5273 ---
[INFO] No sources to compile
[INFO]
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ gh5273 ---
[INFO] No tests to run.
[INFO]
[INFO] --- maven-dependency-plugin:3.1.2:copy-dependencies (copy-libs) @ gh5273 ---
[INFO] Copying helidon-nima-webserver-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-nima-webserver-4.0.0-ALPHA2.jar
[INFO] Copying helidon-common-http-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-common-http-4.0.0-ALPHA2.jar
[INFO] Copying helidon-common-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-common-4.0.0-ALPHA2.jar
[INFO] Copying helidon-common-buffers-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-common-buffers-4.0.0-ALPHA2.jar
[INFO] Copying helidon-common-mapper-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-common-mapper-4.0.0-ALPHA2.jar
[INFO] Copying helidon-common-media-type-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-common-media-type-4.0.0-ALPHA2.jar
[INFO] Copying helidon-common-uri-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-common-uri-4.0.0-ALPHA2.jar
[INFO] Copying helidon-common-parameters-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-common-parameters-4.0.0-ALPHA2.jar
[INFO] Copying helidon-common-socket-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-common-socket-4.0.0-ALPHA2.jar
[INFO] Copying helidon-common-key-util-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-common-key-util-4.0.0-ALPHA2.jar
[INFO] Copying helidon-common-configurable-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-common-configurable-4.0.0-ALPHA2.jar
[INFO] Copying helidon-common-context-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-common-context-4.0.0-ALPHA2.jar
[INFO] Copying helidon-common-config-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-common-config-4.0.0-ALPHA2.jar
[INFO] Copying helidon-nima-common-tls-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-nima-common-tls-4.0.0-ALPHA2.jar
[INFO] Copying helidon-config-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-config-4.0.0-ALPHA2.jar
[INFO] Copying helidon-logging-jul-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-logging-jul-4.0.0-ALPHA2.jar
[INFO] Copying helidon-logging-common-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-logging-common-4.0.0-ALPHA2.jar
[INFO] Copying helidon-nima-http-media-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-nima-http-media-4.0.0-ALPHA2.jar
[INFO] Copying helidon-nima-http-encoding-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-nima-http-encoding-4.0.0-ALPHA2.jar
[INFO] Copying jackson-databind-2.13.2.2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\jackson-databind-2.13.2.2.jar
[INFO] Copying jackson-annotations-2.13.2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\jackson-annotations-2.13.2.jar
[INFO] Copying jackson-core-2.13.2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\jackson-core-2.13.2.jar
[INFO] Copying helidon-graal-native-image-extension-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-graal-native-image-extension-4.0.0-ALPHA2.jar
[INFO] Copying helidon-config-mp-4.0.0-ALPHA2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\helidon-config-mp-4.0.0-ALPHA2.jar
[INFO] Copying jakarta.annotation-api-2.0.0.jar to E:\code\me\reproducers\nima\gh5273\target\libs\jakarta.annotation-api-2.0.0.jar
[INFO] Copying microprofile-config-api-3.0.1.jar to E:\code\me\reproducers\nima\gh5273\target\libs\microprofile-config-api-3.0.1.jar
[INFO] Copying parsson-1.0.2.jar to E:\code\me\reproducers\nima\gh5273\target\libs\parsson-1.0.2.jar
[INFO] Copying jakarta.json-api-2.0.1.jar to E:\code\me\reproducers\nima\gh5273\target\libs\jakarta.json-api-2.0.1.jar
[INFO] Copying classgraph-4.8.87.jar to E:\code\me\reproducers\nima\gh5273\target\libs\classgraph-4.8.87.jar
[INFO]
[INFO] --- helidon-maven-plugin:3.0.0:report (third-party-license-report) @ gh5273 ---
[INFO] Scanning gh5273
[INFO] Reading input from HELIDON_THIRD_PARTY_LICENSES.xml on classpath
[INFO] Writing output to E:\code\me\reproducers\nima\gh5273\target\HELIDON_THIRD_PARTY_LICENSES.txt
[INFO]
[INFO] --- maven-jar-plugin:3.0.2:jar (default-jar) @ gh5273 ---
[INFO] Building jar: E:\code\me\reproducers\nima\gh5273\target\gh5273.jar
[INFO]
[INFO] --- helidon-maven-plugin:3.0.0:native-image (native-image) @ gh5273 ---
[INFO] Building native image :E:\code\me\reproducers\nima\gh5273\target\gh5273
[WARNING] Warning: Ignoring server-mode native-image argument --no-server.
[WARNING] WARNING: Using incubator modules: jdk.incubator.concurrent, jdk.incubator.vector
[INFO] ========================================================================================================================
[INFO] GraalVM Native Image: Generating 'gh5273' (executable)...
[INFO] ========================================================================================================================
[INFO] [1/7] Initializing...                                                                                    (6.4s @ 0.11GB)
[INFO]  Version info: 'GraalVM 22.3.0 Java 19 CE'
[INFO]  Java version info: '19.0.1+10-jvmci-22.3-b08'
[INFO]  C compiler: cl.exe (microsoft, x64, 19.29.30133)
[INFO]  Garbage collector: Serial GC
[INFO]  1 user-specific feature(s)
[INFO]  - io.helidon.integrations.graal.nativeimage.extension.HelidonReflectionFeature
[INFO] 2022.11.03 02:03:35 INFO io.helidon.logging.jul.JulProvider Thread[#1,main,5,main]: Logging at initialization configured using classpath: /logging.properties
[INFO] [2/7] Performing analysis...  [**********]                                                              (19.0s @ 0.53GB)
[INFO]    7,564 (87.73%) of  8,622 classes reachable
[INFO]    9,961 (57.88%) of 17,211 fields reachable
[INFO]   34,505 (55.02%) of 62,715 methods reachable
[INFO]      378 classes,   117 fields, and 1,598 methods registered for reflection
[INFO]       76 classes,    57 fields, and    68 methods registered for JNI access
[INFO]        5 native libraries: crypt32, ncrypt, psapi, version, winhttp
[INFO] [3/7] Building universe...                                                                               (3.5s @ 0.95GB)
[INFO] [4/7] Parsing methods...      [**]                                                                       (2.4s @ 1.08GB)
[INFO] [5/7] Inlining methods...     [***]                                                                      (1.1s @ 0.45GB)
[INFO] [6/7] Compiling methods...    [***]                                                                     (11.6s @ 0.82GB)
[INFO] [7/7] Creating image...                                                                                  (3.0s @ 0.72GB)
[INFO]   14.42MB (47.34%) for code area:    21,828 compilation units
[INFO]   15.74MB (51.69%) for image heap:  209,203 objects and 16 resources
[INFO]  305.00KB ( 0.98%) for other data
[INFO]   30.46MB in total
[INFO] ------------------------------------------------------------------------------------------------------------------------
[INFO] Top 10 packages in code area:                               Top 10 object types in image heap:
[INFO]    1.60MB sun.security.ssl                                     3.03MB byte[] for code metadata
[INFO]  928.56KB java.util                                            1.89MB java.lang.String
[INFO]  753.02KB java.lang.invoke                                     1.74MB java.lang.Class
[INFO]  720.51KB com.sun.crypto.provider                              1.74MB byte[] for general heap data
[INFO]  468.65KB sun.security.x509                                    1.39MB byte[] for java.lang.String
[INFO]  466.68KB io.helidon.config                                  650.03KB com.oracle.svm.core.hub.DynamicHubCompanion
[INFO]  413.53KB java.lang                                          365.50KB byte[] for reflection metadata
[INFO]  394.87KB java.io                                            362.77KB java.util.HashMap$Node
[INFO]  391.88KB java.util.concurrent                               351.17KB java.lang.String[]
[INFO]  275.75KB java.text                                          321.19KB java.lang.Object[]
[INFO]    7.95MB for 224 more packages                                3.49MB for 2072 more object types
[INFO] ------------------------------------------------------------------------------------------------------------------------
[INFO]                        12.7s (24.9% of total time) in 162 GCs | Peak RSS: 2.85GB | CPU load: 5.59
[INFO] ------------------------------------------------------------------------------------------------------------------------
[INFO] Produced artifacts:
[INFO]  E:\code\me\reproducers\nima\gh5273\target\gh5273.build_artifacts.txt (txt)
[INFO]  E:\code\me\reproducers\nima\gh5273\target\gh5273.exe (executable)
[INFO] ========================================================================================================================
[INFO] Finished generating 'gh5273' in 50.3s.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  55.664 s
[INFO] Finished at: 2022-11-03T02:04:20+08:00
[INFO] ------------------------------------------------------------------------
```

**Comment by danielkec** [comment]
Ha! its intermittent, but I can reproduce it

**Comment by danielkec** [comment]
Problem seems to be reproducible only with PowerShell cmd `Invoke-WebRequest`:
```powershell
Invoke-WebRequest -UseBasicParsing -uri "http://localhost:8080/info" -DisableKeepAlive -TimeoutSec 5
```
Helidon accepts connection but no response is sent:
```powershell
2022.11.29 22:23:16 FINE io.helidon.nima.webserver.ConnectionHandler VirtualThread[#200]/runnable@ForkJoinPool-6-worker-37: [0x2611b450 0x2252c228] accepted socket from 127.0.0.1
```
Standard curl works as expected, after eavesdropping with wireshark reproducing the issue with exactly same content doesn't reproduce the issue:
```bash
printf "%s\r\n" \
    "GET /info HTTP/1.1" \
    "User-Agent: Mozilla/5.0 (Windows NT; Windows NT 10.0; en-US) WindowsPowerShell/5.1.19041.1682" \
    "Host: localhost:8080" \
    "Connection: Close" \
    "" |
nc 127.0.0.1 8080

printf "%s\r\n" \
    "GET /info HTTP/1.1" \
    "Host: 127.0.0.1:8080" \
    "User-Agent: curl/7.68.0" \
    "Accept: */*" \
    "" |
nc 127.0.0.1 8080
```

**Comment by danielkec** [comment]
Helidon-less reproducer, works on hotspot for `curl` and `Invoke-WebRequest`, freezes on native-image for  `Invoke-WebRequest` more and `curl` less intermittently:
```java
        ServerSocket serverSocket = new ServerSocket();
        serverSocket.setOption(StandardSocketOptions.SO_REUSEADDR, true);
        serverSocket.setOption(StandardSocketOptions.SO_RCVBUF, 4096);
        serverSocket.bind(new InetSocketAddress("localhost", 8081));
        while (true) {
            Socket socket = serverSocket.accept();
            CompletableFuture.runAsync(() -> {
                        try (Scanner scanner = new Scanner(socket.getInputStream(), StandardCharsets.UTF_8)) {
                            Scanner s = scanner.useDelimiter(DLM);
                            while (s.hasNext()) {
                                var token = s.next();
                                if (token.equals("")) break;
                                System.out.println(token);
                            }

                            OutputStream os = socket.getOutputStream();
                            os.write("HTTP/1.1 200 OK\r\n".getBytes());
                            os.write("Date: Tue, 29 Nov 2022 21:14:59 +0100\r\n".getBytes());
                            os.write("Connection: Close\r\n".getBytes());
                            os.write("Content-Length: 24\r\n".getBytes());
                            os.write("Content-Type: application/json\r\n".getBytes());
                            os.write("\r\n".getBytes());
                            os.write("{\"info\":\"This is info!\"}".getBytes());
                            socket.getOutputStream().flush();
                            socket.getInputStream().close();
                        } catch (IOException e) {
                            throw new RuntimeException(e);
                        }
                    }, Executors.newVirtualThreadPerTaskExecutor())
                    .whenComplete((unused, throwable) -> {
                        try {
                            socket.close();
                        } catch (IOException e) {
                            throw new RuntimeException(e);
                        }
                    });
        }
```

**Comment by danielkec** [comment]
Also reproducible on Linux with way smaller probability(1:1200-4000)
```
5.4.0-132-generic #148~18.04.1-Ubuntu 
[INFO]  Version info: 'GraalVM 22.3.0 Java 19 CE'
[INFO]  Java version info: '19.0.1+10-jvmci-22.3-b08'
[INFO]  C compiler: gcc (linux, x86_64, 7.5.0)
[INFO]  Garbage collector: Serial GC
[INFO]  1 user-specific feature(s)
[INFO]  - io.helidon.integrations.graal.nativeimage.extension.HelidonReflectionFeature
```

**Comment by danielkec** [comment]
Native-image issue reported to GraalVM project oracle/graal/issues/5540

**Comment by danielkec** [comment]
Issue oracle/graal/issues/5540 is fixed in Graal by oracle/graal/pull/5656
